core/arch/x86/mangle.c

```diff
@@ -3858,18 +3858,24 @@ sandbox_top_of_bb(dcontext_t *dcontext, instrlist_t *ilist,
                                  OPND_CREATE_INTPTR(end_pc - (start_pc + 1))));
         PRE(ilist, instr,
             INSTR_CREATE_jcc(dcontext, OP_jge, opnd_create_instr(forward)));
+        /* i#2155: We start at end_pc - 1 the comparison.
+         * Current basic block is [start_pc:end_pc[ = [start_pc:end_pc-1].
+         * This means that end_pc is not included in the basic block.
+         * Hence it must not be compared here.
+         */
         PRE(ilist, instr,
             INSTR_CREATE_mov_imm(dcontext, opnd_create_reg(REG_XDI),
                                  /* will become copy end */
```

Comment:
"will become copy end" needs updating: it won't become equal to the end, but to -1